#164693985 fix validation issue while registration

Chrismarcel opened the pull request:
#### What does this PR do?
Fix validation issues that occur during user signup

#### Description of Task to be completed?
- Write unit tests
- Validate user details

#### How should this be manually tested?
1. Clone this branch.
2. Run `npm install` to install the dependencies.
3. Run `npm test` to run the tests.

#### What are the relevant pivotal tracker stories?
[#164693985](https://pivotaltracker.com/story/show/164693985)

Chrismarcel committed:
bug(user): fix validation issue while registration

- ensure that user details are properly validated

[Finishes #164693985]

danprocoder commented on server/middlewares/ValidateUser.js:
```diff
@@ -27,11 +27,34 @@ class ValidateUser {
+        .exists()
+        .withMessage('Username field must be specified.')
+        .isLength({ min: 3 })
+        .withMessage('Username must not be less than 3 characters.')
```
Excellent job! But username should contain only letters(a-z, A-Z), (no space characters, no numbers, etc). :grin:

Chrismarcel committed:
bug(user): fix user details

- fix length of username
- specify username to be alphabetical only

[Finishes #164693985]

Chrismarcel committed:
fix failing build
